packages/loader/container-loader/src/serializedStateManager.ts

```diff
@@ -66,9 +68,12 @@ export interface IPendingDetachedContainerState extends SnapshotWithBlobs {
 }
 
 export class SerializedStateManager {
-	private readonly processedOps: ISequencedDocumentMessage[] = [];
+	private processedOps: ISequencedDocumentMessage[] = [];
 	private snapshot: SnapshotWithBlobs | undefined;
 	private readonly mc: MonitoringContext;
+	private latestSnapshot: ISnapshotTree | undefined = undefined;
```

Comment:
Maybe I could keep all latest snapshot info in one object, but they are gotten at different times.

Reply:
do it all as one object, and build the whole thing in a single place, even if multiple calls, as we never want partial state.